### ISSUE TITLE
Application to multiple classifications

### ISSUE THREAD

**Issue by Orange2107** [njmarko/gnn-polypharmacy-ddi#2]
Hi @njmarko 
Have you tried applying the model to a multiclassification of 964 side effects?

**Comment by njmarko** [comment]
Hi @Orange2107 

I haven't tried that yet because I currently don't have the hardware to perform such experiments.

It is possible to do something like that, and the authors of the [Drug-Drug Adverse Effect Prediction with Graph Co-Attention](https://arxiv.org/pdf/1905.00534.pdf) paper tried it with their model on the same dataset.

I tried my model on the same dataset with their dataloader, and with some modifications, I believe it could be tweaked to work for multiclassification of side effects. You can check out how the authors of that paper managed to do it for their model in [their code](https://github.com/andreeadeac22/graph_coattention).

**Comment by Orange2107** [comment]
Hi @njmarko 
I'm glad you were able to reply.
I couldn't find a section on multi- classification after looking up her code.
Thank you

**Comment by njmarko** [comment]
I did a quick search in that repo, and I also couldn't find it.

I only found [cross entropy](https://github.com/andreeadeac22/graph_coattention/blob/23781fedaa942ca5614054f965cb7b6543e533fa/graph_nn-master/main.py#L153) in the embedded repo.

It is mentioned in the paper though.

![image](https://github.com/njmarko/gnn-polypharmacy-ddi/assets/34657562/d6ae2821-9407-43ca-aa5d-c7046a73c273)
![image](https://github.com/njmarko/gnn-polypharmacy-ddi/assets/34657562/f98e143c-b6ec-415a-81cd-a0669e325b10)

**Comment by Orange2107** [comment]
@njmarko 
Yeah, I followed this and turned the output of your model into 964 dimensions. I removed the neg set and used the pos set for multiclassification. I get high AUROC but very low AUPRC.
I'm trying to figure that out now.
